components/scream/cime_config/buildnml

```diff
@@ -706,40 +702,57 @@ def create_input_files(caseroot, screamroot, rundir):
         else:
             expect(False, "Unknown input file format '{}'".format(file_format))
 
-        result[file.attrib["name"]] = dict_contents
+        result[file_.attrib["name"]] = dict_contents
 
     return result
 
 ###############################################################################
-def create_input_data_list_file(case,caseroot,screamroot,ic_section):
+def get_file_parameters(caseroot):
 ###############################################################################
+    """
+    Find all XML elements with file="true" attribute. Returns a list
+    of file paths.
+    """
+    raw_xml_file = os.path.join(caseroot, "namelist_scream.xml")
+    with open(raw_xml_file, "r") as fd:
+        tree    = ET.parse(fd)
+        raw_xml = tree.getroot()
 
+    result = []
+    for item in raw_xml.findall('.//*[@type="file"]'):
+        result.append(item.text.strip())
+
+    for item in raw_xml.findall('.//*[@type="array(file)"]'):
+        result.extend(refine_type(item.text, force_type="array(file)"))
+
+    # Remove duplicates. Note sure if an error would be warranted if dupes exist
+    return list(OrderedDict.fromkeys(result))
+
+###############################################################################
+def create_input_data_list_file(case, caseroot):
+###############################################################################
+    """
+    Create the scream.input_data_list file for this case. This will tell CIME
+    what to download.
+    """
     rundir   = case.get_value("RUNDIR")
     target   = os.path.join(rundir, "data")
 
+    files_to_download = get_file_parameters(caseroot)
+
     input_data_list_file = "{}/Buildconf/scream.input_data_list".format(caseroot)
     if os.path.exists(input_data_list_file):
         os.remove(input_data_list_file)
 
-    def_xml_file = os.path.join(screamroot, "cime_config/namelist_defaults_scream.xml")
-    with open(def_xml_file, "r") as fd:
-        tree = ET.parse(fd)
-        root = tree.getroot()
-        tables = get_child(root,"input_files").text.split(',')
-
     with open(input_data_list_file, "w") as fd:
-        fd.write("ic_file_0 = {}\n".format(ic_section["Filename"]))
-        for idx, table in enumerate(tables):
-            fd.write("table_{} = {}\n".format(idx, table.strip()))
-
-    # SCREAM is currently hardcoded to expect tables to be in $rundir/data
-    for table in tables:
-        src_table = do_cime_vars(table.strip(),case)
-        table_name = os.path.basename(src_table)
-        tgt_table = os.path.join(target, table_name)
-        if not os.path.exists(tgt_table) and not os.path.islink(tgt_table):
-            os.symlink(src_table, tgt_table)
+        for idx, file_path in enumerate(files_to_download):
+            fd.write("scream_dl_input_{} = {}\n".format(idx, file_path))
 
+            # SCREAM is currently hardcoded to expect many inputs to be in $rundir/data
+            file_name = os.path.basename(file_path)
```

Comment:
Is this still necessary? I think we used to put the tables in `run/data` when we had them stored in our repo, but now that we get them from the machine's input data folder, we might not need this anymore. If that's the case, I would remove this altogether (and leave our run/data folder skinnier).

Reply:
You're right. Just ran a test and this apparently is not needed anymore.